tests/functional/init/test_init.py

```diff
@@ -84,6 +85,12 @@ def test_init_task_in_project_with_existing_profiles_yml(
 """
             )
 
+    def test_init_task_in_project_specifying_profile_errors(self):
+        # This triggers a call to sys.exit(), requring the following to test it
+        with pytest.raises(SystemExit) as error:
```

Comment:
we have a builtin flag for this

`run_dbt(..., expect_pass=False)`

Reply:
Checking the code for run_dbt() shows that expect_pass only applies to the reported status of the task, not any generated exceptions. Those get reraised.